command/run.go

```diff
@@ -144,6 +145,44 @@ func (c *RunCommand) Synopsis() string {
 
 // Run executes the command.
 func (c *RunCommand) Run(args []string) int {
+	var tmp string
+	var err error
+
+	// Default teeWriter just prints to stdout
+	var teeWriter io.Writer = os.Stdout
+
+	// Create a temporary directory and logfile
+	if !c.dryrun {
+		if tmp, err = util.CreateTemp(); err != nil {
+			fmt.Println("Failed to create temp directory. error:", err)
+			return 1
+		}
+
+		// remove the temporary directory and its contents
+		defer func() {
```

Comment:
OK, so this might be a little weird, but what if CreateTemp returned a cleanup function as well. So, we had something like

```go
if tmp, cleanup, err = util.CreateTemp(); err != nil {
// ...
}

defer cleanup()
```

Reply:
Love that; very clever.

Reply:
Cool! I learned it from tests that set up temporary stuff that needs cleaned up, but it seemed to fit here too!